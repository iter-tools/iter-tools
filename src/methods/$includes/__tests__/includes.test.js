/**
 * @generated-from ./$includes.test.js
 * This file is autogenerated from a template. Please do not edit it directly.
 * To rebuild it from its template use the command
 * > npm run generate
 * More information can be found in CONTRIBUTING.md
 */

/* eslint-disable no-unused-vars,import/no-duplicates */

import { includes, range } from '../../..';
describe('includes', () => {
  it('returns true if the iterable contains the given value', () => {
    expect(includes(1, range(0, 10))).toBe(true);
  });
  it('returns true if the iterable contains only the given value', () => {
    expect(includes(1, [1])).toBe(true);
  });
  it('returns false if the iterable does not contain the given value', () => {
    expect(includes(1, [2])).toBe(false);
  });
  it('returns false if the iterable is empty', () => {
    expect(includes(undefined, [])).toBe(false);
  });
});
