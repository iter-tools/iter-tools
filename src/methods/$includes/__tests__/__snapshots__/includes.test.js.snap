// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`includes when iterable is a string warns 1`] = `
Array [
  Array [
    "For string inputs use includesSubseq instead of includes",
  ],
]
`;
